### PR TITLE
Fixes several warnings

### PR DESCRIPTION
**Pull request by rajithv**
Fixes warnings of the sort:

```
warning: moving a temporary object prevents
      copy elision [-Wpessimizing-move]
    return make_rcp<const UIntPoly>(a.get_var(), std::move(res * tmp));
                                                 ^
```
Fixes some formatting.

### CODE REVIEW

File: symengine/fields.h [hunk 1]
```diff
@@ -91,8 +91,10 @@ class GaloisFieldDict
     // Returns whether polynomial is squarefield in `modulo_`
     bool gf_is_sqf() const;
 
-    // Returns the square free decomposition of polynomial's monic representation in `modulo_`
-    // A vector of pair is returned where each element is a factor and each pair's first
+    // Returns the square free decomposition of polynomial's monic
+    // representation in `modulo_`
+    // A vector of pair is returned where each element is a factor and each
+    // pair's first
     // raised to power of second gives the factor.
```
Comment (nishnik) [inline symengine/fields.h]: Could you change this to 
`// pair's first raised to power of second gives the factor.`
Looks cleaner.

**Comment by nishnik** [comment]
+1 for the `std::move` changes.

**Comment by Sumith1896** [comment]
@rajithv Sorry there seems to conflicts because of a recent PR merge. Could you fix that?
We can then merge this in.

**Comment by certik** [comment]
+1

**Comment by abhinavagarwal07** [comment]
+1

**Comment by Sumith1896** [comment]
@rajithv ping

**Comment by rajithv** [comment]
didn't notice.. thought this was merged. Will fix it.

**Pull request merged**